### PR TITLE
Initial import from the private TL repo.

### PR DESCRIPTION
**Pull request by ereichert**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    description                           := "A set of opinionated SBT plugins for common build and release tasks.",
+    licenses                              += ("Apache-2.0", url("https://www.apache.org/licenses/LICENSE-2.0.html")),
+    publishMavenStyle                     := false,
+    bintrayRepository                     := pluginName,
+    bintrayOrganization in bintray        := None,
+    bintrayReleaseOnPublish in ThisBuild  := false,
+    scalaVersion                          := "2.10.6",
+    scalacOptions                         := Seq("-deprecation", "-feature", "-encoding", "utf8"),
+    resolvers                             += "Typesafe Maven Releases" at "http://repo.typesafe.com/typesafe/releases/",
+    remoteGitRepoPatterns                 ++= Seq("""^git@github.com:ereichert/.*\.git""".r, """^https://github.com/ereichert/.*\.git""".r),
```
Comment (ryandbair) [inline build.sbt]: Should be trafficland, no?

File: src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.NetRCCredentialsProvider;
+import org.eclipse.jgit.transport.URIish;
+
+/**
+ * Interacts with the user during authentication by using the text console.
+ *
+ * @since 4.0
+ */
+public class ConsoleCredentialsProvider extends CredentialsProvider {
```
Comment (ryandbair) [inline src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java]: Oh, interesting. Does this actually work with SBT? 

It looks like this is included in the org.eclipse.jgit.pgm jar. It took a little digging, but I finally found reference to their existence: http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22org.eclipse.jgit%22%20AND%20a%3A%22org.eclipse.jgit.pgm%22

That should be included in plugins.sbt.

File: src/main/resources/init.sh.template [hunk 3]
```diff
@@ -0,0 +1,137 @@
+#!/bin/sh
```
Comment (ryandbair) [inline src/main/resources/init.sh.template]: We can get rid of this. Native packager brings in it's own template.

File: src/main/resources/nginx.conf.template [hunk 4]
```diff
@@ -0,0 +1,51 @@
+server {
```
Comment (ryandbair) [inline src/main/resources/nginx.conf.template]: This file is also unused.

**Review by ryandbair — CHANGES_REQUESTED** [review]

**Comment by ereichert** [comment]
@ryandbair In general, this isn't ready for review. I should have deleted the PR.

File: build.sbt (hunk 1 above)
Comment (ereichert) [reply build.sbt]: Once I finish testing, yes.

File: src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java (hunk 2 above)
Comment (ereichert) [reply src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java]: Yes. It works as it is. The plugin however does not. That class never made it to release as far as I can tell. We only need those two bits of code so copying the code seems reasonable.

File: src/main/resources/init.sh.template (hunk 3 above)
Comment (ereichert) [reply src/main/resources/init.sh.template]: 10-4

File: src/main/resources/nginx.conf.template (hunk 4 above)
Comment (ereichert) [reply src/main/resources/nginx.conf.template]: 10-4

**Comment by ereichert** [comment]
@ryandbair This is finally ready to be reviewed.

**Comment by ereichert** [comment]
@ryandbair or @cchampignon Could one of you finish this up?

**Comment by ryandbair** [comment]
Will do.

File: project/KeyGenerator.scala [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import sbt._
+
+object KeyGenerator {
+
+    val generateKeysObject: TaskKey[Seq[File]] = TaskKey[Seq[File]]("generate-keys-object", "Generates the Keys Object which aliases all the plugins' keys in one object.")
+    val keysFile: SettingKey[File] = SettingKey[File]("keys-file", "Keys file to generate")
+
+    def writeKeysObject(targetFile: File): Seq[File] = {
+      targetFile.getParentFile.mkdirs()
+      IO.write(targetFile, "bin/generate-keys".!!)
```
Comment (ryandbair) [inline project/KeyGenerator.scala:10]: generate-keys and generate-keys.rb aren't checked in. 

Long term, I think it makes sense to reorganize the keys into different traits with companion objects and combine the traits into larger objects, similar to what other sbt plugins are doing.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+**Before you do anything else,
```
Comment (ryandbair) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I never knew this was a thing. Nifty.

File: src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java (hunk 2 above)
Comment (ryandbair) [reply src/main/java/com/trafficland/augmentsbt/ConsoleCredentialsProvider.java]: I have a commit which fixes this. PR inbound.

**Review by ryandbair — CHANGES_REQUESTED** [review]

File: project/KeyGenerator.scala (hunk 5 above)
Comment (ereichert) [reply project/KeyGenerator.scala:10]: Resolved.

**Comment by ereichert** [comment]
@ryandbair This includes your PR. It's ready to be merged.

**Review by ryandbair — APPROVED** [review]

**Pull request merged**